Library/lightconf/config_format.hpp

```diff
@@ -1,6 +1,11 @@
 #ifndef _LIGHTCONF_OUTER_CONFIG_FORMAT_H_
 #define _LIGHTCONF_OUTER_CONFIG_FORMAT_H_
 
+#pragma GCC diagnostic push
```

Comment:
Don't put that here. Create a header file which sole purpose is to include external headers. And there, do the pragmas. It'll make things easier in the future.